Problem2.py

```diff
@@ -0,0 +1,80 @@
+'''
+DFS:
+1. We get the level of the node and the parent of the nodes whose values are x and y depth-wise. 
+2. If the level of the nodes are equal and the parents are not equal, then we return True.
+3. Else we return False.
+
+TC: O(N)
+SC: O(H) - Recursive stack 
+
+BFS:
+1. We traverse level by level and check if the current node's left and right child are equal to x and y. If yes then they aren't cousins.
+2. If the current node's value is equal to x or y, then we set the respective flag to True.
+3. If both the flags are True, then we return True. If only one of them is found, then we return False.
+
+TC: O(N)
+SC: O(N) - Diameter of the tree
```

Comment:
What do you mean by diameter here

Reply:
Diameter is the number of nodes in the last level of the tree.